### DISCUSSION TITLE
Backup/Restore cluster config

### DISCUSSION THREAD

**Discussion by rpiteira** [fluxcd/flux2#3023] [Q&A] [answered]
Hi all,

I have the following atypical scenario: Every development team will be responsable of creating a flux config to sync the configmaps, images, deployments, etc of each application.

When the infrastructure team needs to rebuild the cluster, how they can restore the config without asking the teams to recreate the flux flows?

I was thinking of Velero, but I was wondering if there is a simpler solution?

Thanks a lot for your help

**Comment by stefanprodan** [comment, marked as answer]
If you’re using this repo structure https://github.com/fluxcd/flux2-multi-tenancy then cluster admin will just rerun bootstrap on the next cluster using the same path and everything will be restored from Git.

**Reply by kingdonb** [reply]
https://github.com/fluxcd/flux2-multi-tenancy * fixed link

If you have a need for stateful workloads, we intend to cover this topic at GitOpsCon this year:
https://gitopsconna22.sched.com/event/1AR9B/complete-dr-of-stateful-workloads-pvs-and-csi-snapshots-via-flux-and-vault-oss-vanessa-abankwah-kingdon-barrett-weaveworks

The short answer is, persistent volumes can be snapshotted, and Velero is one way to accomplish this. (But then what do you do with the snapshots? ... come see the talk!)

**Reply by rpiteira** [reply]
Thanks a lot, I finally decided to create a script to build all the flux create commands for the new cluster before destroying it! Easier I think.
